automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoApiClient.scala

```diff
@@ -538,7 +538,7 @@ object LeonardoApiClient {
   def listApps(
     googleProject: GoogleProject,
     includeDeleted: Boolean = false
-  )(implicit client: Client[IO], authHeader: Authorization): IO[List[ListAppResponse]] = {
+  )(implicit client: Client[IO], authHeader: => Authorization): IO[List[ListAppResponse]] = {
```

Comment:
I thought we talked about making this an `IO[Authorization]`? Curious why "by name" was chosen.

I _think_ by-name still works but I had to test it:

```
// Make a side effecting function
scala> def sideEffect(): String = {
     | println("side effect")
     | "side effect"
     | }
def sideEffect(): String

// Wraps a String in an IO by value
scala> def byValue(s: String): IO[String] = IO(s)
def byValue(s: String): cats.effect.IO[String]

// Side effect is done only once when the IO is built
scala> val io = byValue(sideEffect())
side effect
val io: cats.effect.IO[String] = IO(...)

scala> io.unsafeRunSync()
val res56: String = side effect

scala> io.unsafeRunSync()
val res57: String = side effect

scala> io.unsafeRunSync()
val res58: String = side effect

// Wrap a string in an IO by name
scala> def byName(s: => String): IO[String] = IO(s)
def byName(s: => String): cats.effect.IO[String]

scala> val io = byName(sideEffect())
val io: cats.effect.IO[String] = IO(...)

// Side effect is done everytime the IO is _run_
scala> io.unsafeRunSync()
side effect
val res59: String = side effect

scala> io.unsafeRunSync()
side effect
val res60: String = side effect

scala> io.unsafeRunSync()
side effect
val res61: String = side effect
```

Reply:
think I like `IO[Authorization]` better 🤔